presto-main/src/main/java/com/facebook/presto/spiller/LocalTempStorage.java

```diff
@@ -119,6 +122,20 @@ public void remove(TempDataOperationContext context, TempStorageHandle handle)
         Files.delete(((LocalTempStorageHandle) handle).getFilePath());
     }
 
+    @Override
+    public InputStream open(TempDataOperationContext context, URI fileURI)
```

Comment:
Sure.. The problem with storage handle was that it cannot be serialized. I fixed it by adding a new API that accepts URI which is serializable (under the assumption that that every storage will have URI). Ideal option was to make StorageHandle serializable which will work for storage implementations

Reply:
`StorageHandle` is just a marker interface. It might be hard to enforce that every implementation is Java serializable. Though I suggest to add 2 explicit methods to the `StorageService` interface to serialize  / deserialize handles.